Dietpi-Software | ZeroTier

AAS-Crypt opened the pull request:
I've added Zerotier, can someone please explain what exactly is "aSOFTWARE_DEPS" and "aSOFTWARE_INTERACTIVE" stands for?
Does Interactive mean that software has website as interacting mediator?

Tested on Raspberry Pi 4 8GB

AAS-Crypt committed:
Adding to software list "ZeroTier"

I've added Zerotier, can someone please explain what exactly is "aSOFTWARE_DEPS" and "aSOFTWARE_INTERACTIVE" stands for?

Joulinar commented on dietpi/dietpi-software:
```diff
@@ -8125,7 +8132,26 @@ _EOF_
+                        [[ -f '/usr/sbin/zerotier-cli' ]] && G_EXEC rm /usr/sbin/zerotier-cli
+                        [[ -f '/usr/sbin/zerotier-one' ]] && G_EXEC rm /usr/sbin/zerotier-one
+                        [[ -f '/etc/apt/sources.list.d/zerotier.list' ]] && G_EXEC rm /etc/apt/sources.list.d/zerotier.list
+                        [[ -f '/var/lib/zerotier-one/identity.public' ]] && G_EXEC rm /var/lib/zerotier-one/identity.public
```
This might not be needed if we setup the source list and public key directly. Usually this will not change and could stay on a reinstall.

Joulinar commented on dietpi/dietpi-software:
```diff
@@ -8125,7 +8132,26 @@ _EOF_
+                        [[ -f '/etc/apt/sources.list.d/zerotier.list' ]] && G_EXEC rm /etc/apt/sources.list.d/zerotier.list
+                        [[ -f '/var/lib/zerotier-one/identity.public' ]] && G_EXEC rm /var/lib/zerotier-one/identity.public
+                        [[ -f '/etc/init.d/zerotier-one' ]] && G_EXEC rm /etc/init.d/zerotier-one
+			G_EXEC apt remove zerotier-one -y
```
Usually we don't purge software before duing a reinstall

Joulinar commented on dietpi/dietpi-software:
```diff
@@ -8125,7 +8132,26 @@ _EOF_
+                        [[ -f '/etc/init.d/zerotier-one' ]] && G_EXEC rm /etc/init.d/zerotier-one
+			G_EXEC apt remove zerotier-one -y
+                        #Installation
+			G_EXEC wget -O zerotier-installer.sh https://install.zerotier.com
```
let's try to avoid using the install script. Better to install key and source list on our own. Following could do

			Banner_Installing

			# APT key
			G_EXEC eval "curl -sSLf https://raw.githubusercontent.com/zerotier/ZeroTierOne/master/doc/contact%40zerotier.com.gpg | gpg --dearmor -o /etc/apt/trusted.gpg.d/dietpi-zerotier.gpg --yes"

			# APT list
			G_EXEC eval "echo 'deb https://download.zerotier.com/debian/$G_DISTRO_NAME $G_DISTRO_NAME main' > /etc/apt/sources.list.d/dietpi-zerotier.list"

			G_AGUP
			G_AGI zerotier-one
		fi

Joulinar commented on dietpi/dietpi-software:
```diff
@@ -14205,6 +14231,18 @@ If no WireGuard (auto)start is included, but you require it, please do the follo
+                        [[ -f '/etc/apt/sources.list.d/zerotier.list' ]] && G_EXEC rm /etc/apt/sources.list.d/zerotier.list
+                        [[ -f '/var/lib/zerotier-one/identity.public' ]] && G_EXEC rm /var/lib/zerotier-one/identity.public
+                        [[ -f '/etc/init.d/zerotier-one' ]] && G_EXEC rm /etc/init.d/zerotier-one
+			G_EXEC apt remove zerotier-one -y
```
you would need to purge the package first before removing files. As well, pls use our own function like `G_AGP` instead of `apt remove`

AAS-Crypt replied:
I didn't understand all the shortcuts if there are file with command and descriptions could you please show me?

Joulinar commented on dietpi/dietpi-software:
```diff
@@ -1388,6 +1388,13 @@ Available commands:
+		aSOFTWARE_DESC[$software_id]='Free easy to deploy Cloud-hosted VPN service'
+		aSOFTWARE_CATX[$software_id]=15
+		aSOFTWARE_DOCS[$software_id]='https://dietpi.com/docs/software/vpn/#zerotier'
+		#aSOFTWARE_DEPS[$software_id]='196'
```
This would meand your Software depends on JAVA.

Joulinar commented on dietpi/dietpi-software:
```diff
@@ -1401,6 +1401,12 @@ Available commands:
 		aSOFTWARE_CATX[$software_id]=15
 		aSOFTWARE_DOCS[$software_id]='https://dietpi.com/docs/software/vpn/#tailscale'
+		#-----------------
+		software_id=203
```
```suggestion
		software_id=201
```

Joulinar commented on dietpi/dietpi-software:
```diff
@@ -8226,7 +8232,26 @@ _EOF_
 		fi
-
+		
+                software_id=203 # Zerotier
```
```suggestion
                software_id=201 # Zerotier
```

Joulinar commented on dietpi/dietpi-software:
```diff
@@ -14336,6 +14361,18 @@ If no WireGuard (auto)start is included, but you require it, please do the follo
 
 		fi
 
+                software_id=203 # Zerotier
```
```suggestion
                software_id=201 # Zerotier
```

MichaIng committed:
v8.6

- DietPi-Software | ZeroTier: Do not stop service after identity has been created, but remove SysV service
- DietPi-Software | Jellyfin: Avoid doubled service stop when waiting for config file and do not wait for config file if we do not touch it anyway

MichaIng committed:
v8.6

- DietPi-Software | ZeroTier: Remove data directory on uninstall, not remove by package purge

MichaIng committed:
v8.6

- DietPi-Software | ZeroTier: Add all meta info
